src/blockquotecommand.js

```diff
@@ -223,3 +227,48 @@ function checkCanBeQuoted( schema, block ) {
 
 	return isBQAllowed && isBlockAllowedInBQ;
 }
+
+// Returns top-level block elements from selected blocks.
+//
+// For given selection:
+//
+//		[<blockA></blockA>
+//		<blockB>
+//			<blockC></blockC>
+//			<blockD></blockD>
+//		</blockB>
+//		<blockE></blockE>]
+//
+// This method will only return blocks: A, B & E.
+//
+// @param {module:engine/model/selection~Selection} selection
+// @param {module:engine/model/schema~Schema} schema
+// @returns {module:engine/model/element~Element[]}
+function getTopMostBlocks( selection, schema ) {
```

Comment:
Maybe this should be `Selection` method? I can imagine other features needing this too.